test/NuGet.Core.Tests/NuGet.ProjectModel.Test/LockFileFormatTests.cs

```diff
@@ -1805,5 +1802,85 @@ public void LockFileFormat_ReadsLockFileWithEmbedAssemblies()
             Assert.Equal("lib/netstandard2.0/packageA.interop.dll", targetLibrary.RuntimeAssemblies.Single().Path);
             Assert.Equal("embed/netstandard2.0/packageA.interop.dll", targetLibrary.EmbedAssemblies.Single().Path);
         }
+
+        [Fact]
+        public void LockFileFormat_WritesFrameworkReference()
+        {
+
+            // Arrange
+            var lockFileContent = @"{
+  ""version"": 3,
+  ""targets"": {
+                "".NETCoreApp,Version=v3.0"": {
+                    ""My.Nice.Package.With.WPF.Reference/2.0.0"": {
+                        ""type"": ""package"",
+        ""compile"": {
+                            ""lib/netcoreapp3.0/a.dll"": { }
+                        },
+        ""frameworkReferences"": [
+          ""Microsoft.Windows.Desktop|WPF"",
+          ""Microsoft.Windows.Desktop|WindowsForms""
+        ]
+    }
+}
+  },
+  ""libraries"": {
+    ""My.Nice.Package.With.WPF.Reference/2.0.0"": {
+      ""sha512"": ""sup3rs3cur3"",
+      ""type"": ""package"",
+      ""files"": [
+        ""My.Nice.Package.With.WPF.Reference.nuspec"",
+        ""lib/netcoreapp3.0/a.dll""
+      ]
+    }
+  },
+  ""projectFileDependencyGroups"": { }
+}";
+            var lockFile = new LockFile()
+            {
+                Version = 3
```

Comment:
Why not 4?  Are we verifying anywhere that the lock file format version is now 4 with this PR?

It seems like there should be a single lock file format test that verifies the current version and throws if any unexpected changes are present.

Reply:
I will fix the tests.

Reply:
Actually I just realized that the version kind of doesn't do much. 

I will revert to 3, and create a tracking issue to address this.